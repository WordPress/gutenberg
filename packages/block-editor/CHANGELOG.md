## Master

### Breaking Changes

- `BlockEditorProvider` no longer renders a wrapping `SlotFillProvider` or `DropZoneProvider` (from `@wordpress/components`). For custom block editors, you should render your own as wrapping the `BlockEditorProvider`. A future release will include a new `BlockEditor` component for simple, standard usage. `BlockEditorProvider` will serve the simple purpose of establishing its own context for block editors.

## 2.2.0 (2019-06-12)

### Internal

- Refactored `BlockSettingsMenu` to use `DropdownMenu` from `@wordpress/components`.

## 2.0.0 (2019-04-16)

### New Features

- Added the `addToGallery` property to the `MediaUpload` interface. The property allows users to open the media modal in the `gallery-library`instead of `gallery-edit` state.
- Added the `addToGallery` property to the `MediaPlaceholder` component. The component passes the property to the `MediaUpload` component used inside the placeholder.
- Added the `isAppender` property to the `MediaPlaceholder` component. The property changes the look of the placeholder to be adequate to scenarios where new files are added to an already existing set of files, e.g., adding files to a gallery.
- Added the `dropZoneUIOnly` property to the `MediaPlaceholder` component. The property makes the `MediaPlaceholder` only render a dropzone without any other additional UI.
- Added a cancel link to the list of buttons in the `MediaPlaceholder` component which appears if an `onCancel` handler exists.
- Added the usage of `mediaPreview` for the `Placeholder` component to the `MediaPlaceholder` component.
- Added a an `onDoubleClick` event handler to the `MediaPlaceholder` component.
- Added a way to pass special `ref` property to the `PlainText` component.
- The `URLPopover` component now passes through all unhandled props to the underlying Popover component.

### Breaking Changes

- `CopyHandler` will now only catch cut/copy events coming from its `props.children`, instead of from anywhere in the `document`.

### Internal

- Improved handling of blocks state references for unchanging states.
- Updated handling of blocks state to effectively ignored programmatically-received blocks data (e.g. reusable blocks received from editor).

## 1.0.0 (2019-03-06)

### New Features

- Initial version.
