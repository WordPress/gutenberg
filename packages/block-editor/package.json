{
	"name": "@wordpress/block-editor",
	"version": "2.2.0",
	"description": "Generic block editor.",
	"author": "The WordPress Contributors",
	"license": "GPL-2.0-or-later",
	"keywords": [
		"wordpress",
		"editor",
		"blocks"
	],
	"homepage": "https://github.com/WordPress/gutenberg/tree/master/packages/block-editor/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/WordPress/gutenberg.git",
		"directory": "packages/block-editor"
	},
	"bugs": {
		"url": "https://github.com/WordPress/gutenberg/issues"
	},
	"main": "build/index.js",
	"module": "build-module/index.js",
	"react-native": "src/index",
	"dependencies": {
		"@babel/runtime": "^7.4.4",
		"@wordpress/a11y": "file:../a11y",
		"@wordpress/blob": "file:../blob",
		"@wordpress/blocks": "file:../blocks",
		"@wordpress/components": "file:../components",
		"@wordpress/compose": "file:../compose",
		"@wordpress/data": "file:../data",
		"@wordpress/dom": "file:../dom",
		"@wordpress/element": "file:../element",
		"@wordpress/hooks": "file:../hooks",
		"@wordpress/html-entities": "file:../html-entities",
		"@wordpress/i18n": "file:../i18n",
		"@wordpress/is-shallow-equal": "file:../is-shallow-equal",
		"@wordpress/keycodes": "file:../keycodes",
		"@wordpress/rich-text": "file:../rich-text",
		"@wordpress/token-list": "file:../token-list",
		"@wordpress/url": "file:../url",
		"@wordpress/viewport": "file:../viewport",
		"@wordpress/wordcount": "file:../wordcount",
		"classnames": "^2.2.5",
		"dom-scroll-into-view": "^1.2.1",
		"lodash": "^4.17.10",
		"react-spring": "^8.0.19",
		"redux-multi": "^0.1.12",
		"refx": "^3.0.0",
		"rememo": "^3.0.0",
		"tinycolor2": "^1.4.1"
	},
	"publishConfig": {
		"access": "public"
	}
}
