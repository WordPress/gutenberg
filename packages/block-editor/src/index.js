/**
 * WordPress dependencies
 */
import '@wordpress/blocks';
import '@wordpress/core-data';
import '@wordpress/rich-text';
import '@wordpress/viewport';

/**
 * Internal dependencies
 */
import './store';
import './hooks';

export * from './components';
export * from './utils';
export { storeConfig } from './store';
export { SETTINGS_DEFAULTS } from './store/defaults';
