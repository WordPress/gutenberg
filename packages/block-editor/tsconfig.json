{
	"$schema": "https://json.schemastore.org/tsconfig.json",
	"extends": "../../tsconfig.base.json",
	"compilerOptions": {
		"rootDir": "src",
		"declarationDir": "build-types"
	},
	"references": [
		{ "path": "../a11y" },
		{ "path": "../api-fetch" },
		{ "path": "../blob" },
		{ "path": "../components" },
		{ "path": "../compose" },
		{ "path": "../data" },
		{ "path": "../date" },
		{ "path": "../deprecated" },
		{ "path": "../dom" },
		{ "path": "../element" },
		{ "path": "../escape-html" },
		{ "path": "../private-apis" },
		{ "path": "../hooks" },
		{ "path": "../html-entities" },
		{ "path": "../i18n" },
		{ "path": "../icons" },
		{ "path": "../is-shallow-equal" },
		{ "path": "../keycodes" },
		{ "path": "../notices" },
		{ "path": "../rich-text" },
		{ "path": "../style-engine" },
		{ "path": "../token-list" },
		{ "path": "../url" },
		{ "path": "../warning" },
		{ "path": "../wordcount" }
	],
	// NOTE: This package is being progressively typed. You are encouraged to
	// expand this array with files which can be type-checked. At some point in
	// the future, this can be simplified to an `includes` of `src/**/*`.
	"files": [ "src/components/block-context/index.js", "src/utils/dom.js" ]
}
