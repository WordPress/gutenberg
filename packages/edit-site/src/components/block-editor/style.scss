.edit-site-visual-editor__editor-canvas {
	&.is-focused {
		outline: calc(2 * var(--wp-admin-border-width-focus)) solid var(--wp-admin-theme-color);
		outline-offset: calc(-2 * var(--wp-admin-border-width-focus));
	}
}
