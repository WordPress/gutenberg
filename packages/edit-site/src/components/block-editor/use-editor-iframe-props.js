/**
 * External dependencies
 */
import clsx from 'clsx';

/**
 * WordPress dependencies
 */
import { useSelect, useDispatch } from '@wordpress/data';
import { ENTER, SPACE } from '@wordpress/keycodes';
import { useState, useEffect } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import { store as editorStore } from '@wordpress/editor';

/**
 * Internal dependencies
 */
import { unlock } from '../../lock-unlock';
import { store as editSiteStore } from '../../store';

export default function useEditorIframeProps() {
	const { canvasMode, currentPostIsTrashed } = useSelect( ( select ) => {
		const { getCanvasMode } = unlock( select( editSiteStore ) );

		return {
			canvasMode: getCanvasMode(),
			currentPostIsTrashed:
				select( editorStore ).getCurrentPostAttribute( 'status' ) ===
				'trash',
		};
	}, [] );
	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
	const [ isFocused, setIsFocused ] = useState( false );

	useEffect( () => {
		if ( canvasMode === 'edit' ) {
			setIsFocused( false );
		}
	}, [ canvasMode ] );

	// In view mode, make the canvas iframe be perceived and behave as a button
	// to switch to edit mode, with a meaningful label and no title attribute.
	const viewModeIframeProps = {
		'aria-label': __( 'Edit' ),
		'aria-disabled': currentPostIsTrashed,
		title: null,
		role: 'button',
		tabIndex: 0,
		onFocus: () => setIsFocused( true ),
		onBlur: () => setIsFocused( false ),
		onKeyDown: ( event ) => {
			const { keyCode } = event;
			if (
				( keyCode === ENTER || keyCode === SPACE ) &&
				! currentPostIsTrashed
			) {
				event.preventDefault();
				setCanvasMode( 'edit' );
			}
		},
		onClick: () => {
			setCanvasMode( 'edit' );
		},
		onClickCapture: ( event ) => {
			if ( currentPostIsTrashed ) {
				event.preventDefault();
				event.stopPropagation();
			}
		},
		readonly: true,
	};

	return {
		className: clsx( 'edit-site-visual-editor__editor-canvas', {
			'is-focused': isFocused && canvasMode === 'view',
		} ),
		...( canvasMode === 'view' ? viewModeIframeProps : {} ),
	};
}
