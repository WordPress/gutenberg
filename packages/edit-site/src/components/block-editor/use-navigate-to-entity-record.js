/**
 * WordPress dependencies
 */
import { privateApis as routerPrivateApis } from '@wordpress/router';
import { useCallback } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { unlock } from '../../lock-unlock';

const { useHistory } = unlock( routerPrivateApis );

export default function useNavigateToEntityRecord() {
	const history = useHistory();

	const onNavigateToEntityRecord = useCallback(
		( params ) => {
			history.push( { ...params, focusMode: true, canvas: 'edit' } );
		},
		[ history ]
	);

	return onNavigateToEntityRecord;
}
