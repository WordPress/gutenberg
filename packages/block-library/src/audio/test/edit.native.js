/**
 * External dependencies
 */
import {
	addBlock,
	dismissModal,
	fireEvent,
	getBlock,
	getEditorHtml,
	initializeEditor,
	openBlockSettings,
	render,
	screen,
	setupCoreBlocks,
} from 'test/helpers';

/**
 * WordPress dependencies
 */
import { BlockEdit } from '@wordpress/block-editor';
import {
	subscribeMediaUpload,
	sendMediaUpload,
} from '@wordpress/react-native-bridge';

/**
 * Internal dependencies
 */
import { name } from '../index';

// react-native-aztec shouldn't be mocked because these tests are based on
// snapshot testing where we want to keep the original component.
jest.unmock( '@wordpress/react-native-aztec' );

const MEDIA_UPLOAD_STATE_FAILED = 3;

const AUDIO_BLOCK = `<!-- wp:audio {"id":5} -->
<figure class="wp-block-audio"><audio controls src="https://cldup.com/59IrU0WJtq.mp3"></audio></figure>
<!-- /wp:audio -->`;

let uploadCallBack;
subscribeMediaUpload.mockImplementation( ( callback ) => {
	uploadCallBack = callback;
} );
sendMediaUpload.mockImplementation( ( payload ) => {
	uploadCallBack( payload );
} );

const AudioEdit = ( { clientId, ...props } ) => (
	<BlockEdit name={ name } clientId={ clientId || 0 } { ...props } />
);

const getTestComponentWithContent = ( attributes = {} ) => {
	return render(
		<AudioEdit attributes={ attributes } setAttributes={ jest.fn() } />
	);
};

setupCoreBlocks( [ 'core/audio' ] );

describe( 'Audio block', () => {
	it( 'renders placeholder without crashing', () => {
		const component = getTestComponentWithContent();
		const rendered = component.toJSON();
		expect( rendered ).toMatchSnapshot();
	} );

	it( 'renders audio file without crashing', () => {
		const component = getTestComponentWithContent( {
			src: 'https://cldup.com/59IrU0WJtq.mp3',
			id: '1',
		} );

		const rendered = component.toJSON();
		expect( rendered ).toMatchSnapshot();
	} );

	it( 'renders audio block error state without crashing', () => {
		const MEDIA_ID = '1';
		const component = getTestComponentWithContent( {
			src: 'https://cldup.com/59IrU0WJtq.mp3',
			id: MEDIA_ID,
		} );

		const payloadFail = {
			state: MEDIA_UPLOAD_STATE_FAILED,
			mediaId: MEDIA_ID,
		};
		sendMediaUpload( payloadFail );

		const rendered = component.toJSON();
		expect( rendered ).toMatchSnapshot();
	} );

	it( 'should gracefully handle invalid URLs', async () => {
		await initializeEditor();

		await addBlock( screen, 'Audio' );
		fireEvent.press( screen.getByText( 'Insert from URL' ) );
		fireEvent.changeText(
			screen.getByPlaceholderText( 'Type a URL' ),
			'h://wordpress.org/audio.mp3'
		);
		dismissModal( screen.getByTestId( 'bottom-sheet' ) );

		expect(
			screen.getByText( 'Invalid URL. Audio file not found.' )
		).toBeVisible();
	} );

	it( 'should enable autoplay setting', async () => {
		await initializeEditor( { initialHtml: AUDIO_BLOCK } );

		const audioBlock = getBlock( screen, 'Audio' );
		fireEvent.press( audioBlock );
		await openBlockSettings( screen );

		fireEvent.press( screen.getByText( 'Autoplay' ) );
		expect( getEditorHtml() ).toMatchSnapshot();
	} );

	it( 'should enable loop setting', async () => {
		await initializeEditor( { initialHtml: AUDIO_BLOCK } );

		const audioBlock = getBlock( screen, 'Audio' );
		fireEvent.press( audioBlock );
		await openBlockSettings( screen );

		fireEvent.press( screen.getByText( 'Loop' ) );
		expect( getEditorHtml() ).toMatchSnapshot();
	} );
} );
