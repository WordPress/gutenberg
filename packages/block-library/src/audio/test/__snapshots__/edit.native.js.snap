// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio block renders audio block error state without crashing 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    pointerEvents="box-none"
    style={
      [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        [
          undefined,
          undefined,
          undefined,
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": 44,
        }
      }
    >
      <View>
        <Svg
          height={24}
          style={
            {
              "fill": "gray",
            }
          }
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          Path
        </Svg>
      </View>
      <View
        style={{}}
      >
        <Text>
          59IrU0WJtq
        </Text>
        <View>
          <Svg
            height={16}
            style={{}}
            viewBox="-2 -2 24 24"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            Path
          </Svg>
          <Text
            style={{}}
          >
            Failed to insert audio file. Please tap for options.
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "height": 44,
        },
        false,
      ]
    }
  >
    <View
      accessibilityLabel="Audio caption. Empty"
      accessibilityRole="button"
      accessible={true}
      style={
        {
          "display": "none",
          "flex": 1,
        }
      }
    >
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <RCTAztecView
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          activeFormats={[]}
          blockType={
            {
              "tag": "p",
            }
          }
          deleteEnter={true}
          disableEditingMenu={false}
          focusable={true}
          fontFamily="serif"
          fontSize={14}
          isMultiline={false}
          maxImagesWidth={200}
          onBackspace={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onContentSizeChange={[Function]}
          onEnter={[Function]}
          onFocus={[Function]}
          onHTMLContentWithCursor={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onSelectionChange={[Function]}
          onStartShouldSetResponder={[Function]}
          placeholder="Add caption"
          placeholderTextColor="gray"
          selectionColor="black"
          style={
            {
              "backgroundColor": undefined,
              "maxWidth": undefined,
              "minHeight": 0,
            }
          }
          text={
            {
              "eventCount": undefined,
              "linkTextColor": undefined,
              "selection": null,
              "tag": "p",
              "text": "",
            }
          }
          textAlign="center"
          triggerKeyCodes={[]}
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Audio block renders audio file without crashing 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    pointerEvents="box-none"
    style={
      [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        [
          undefined,
          undefined,
          undefined,
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": 44,
        }
      }
    >
      <View>
        <Svg
          height={24}
          style={
            {
              "fill": "gray",
            }
          }
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          Path
        </Svg>
      </View>
      <View
        style={{}}
      >
        <Text>
          59IrU0WJtq
        </Text>
        <View>
          <Text
            style={{}}
          >
            MP3 audio file
          </Text>
        </View>
      </View>
      <View
        accessibilityHint="Double tap to listen the audio file"
        accessibilityLabel="Audio Player"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          OPEN
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "height": 44,
        },
        false,
      ]
    }
  >
    <View
      accessibilityLabel="Audio caption. Empty"
      accessibilityRole="button"
      accessible={true}
      style={
        {
          "display": "none",
          "flex": 1,
        }
      }
    >
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <RCTAztecView
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          activeFormats={[]}
          blockType={
            {
              "tag": "p",
            }
          }
          deleteEnter={true}
          disableEditingMenu={false}
          focusable={true}
          fontFamily="serif"
          fontSize={14}
          isMultiline={false}
          maxImagesWidth={200}
          onBackspace={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onContentSizeChange={[Function]}
          onEnter={[Function]}
          onFocus={[Function]}
          onHTMLContentWithCursor={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onSelectionChange={[Function]}
          onStartShouldSetResponder={[Function]}
          placeholder="Add caption"
          placeholderTextColor="gray"
          selectionColor="black"
          style={
            {
              "backgroundColor": undefined,
              "maxWidth": undefined,
              "minHeight": 0,
            }
          }
          text={
            {
              "eventCount": undefined,
              "linkTextColor": undefined,
              "selection": null,
              "tag": "p",
              "text": "",
            }
          }
          textAlign="center"
          triggerKeyCodes={[]}
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Audio block renders placeholder without crashing 1`] = `
<View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          [
            [
              undefined,
              undefined,
            ],
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            {
              "fill": "black",
            }
          }
        >
          <View
            style={{}}
          >
            <Svg
              fill="black"
              height={24}
              style={{}}
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              Path
            </Svg>
          </View>
        </View>
        <Text>
          Audio
        </Text>
      </View>
      <View
        accessibilityHint="Double tap to select an audio file"
        accessibilityLabel="Audio block. Empty"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 22,
            "left": 22,
            "right": 22,
            "top": 22,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text>
          Add audio
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Audio block should enable autoplay setting 1`] = `
"<!-- wp:audio {"id":5} -->
<figure class="wp-block-audio"><audio controls src="https://cldup.com/59IrU0WJtq.mp3" autoplay></audio></figure>
<!-- /wp:audio -->"
`;

exports[`Audio block should enable loop setting 1`] = `
"<!-- wp:audio {"id":5} -->
<figure class="wp-block-audio"><audio controls src="https://cldup.com/59IrU0WJtq.mp3" loop></audio></figure>
<!-- /wp:audio -->"
`;
